app/models/event.rb

```diff
@@ -4,4 +4,13 @@ class Event < ActiveRecord::Base
   validates :slug, presence: true, uniqueness:true
   validates :location, presence: true, length: { maximum: 128 }
   validates :date, presence: true
+
+  def self.upcoming
+    where('date > ?', DateTime.now).order(date: :desc)
+  end
+
+  def self.previous
```

Comment:
could define the two functions to take in a limit that defaults to 4 maybe? instead of calling limit on them all the time:

def self.upcoming/previous(limit = 4)